.travis.yml

```diff
@@ -28,6 +30,38 @@ before_script:
   - wget https://raw.githubusercontent.com/torvalds/linux/master/scripts/spelling.txt
   - cd $MYHOME
 
+  # Tools required for QEMU tests
+  # 'apt-get install' cannot be used in the new container-based infrastructure
+  # (which is the only allowing caching), so we just build from sources
+  # bc is used during kernel configuration
+  - cd $HOME
+  - wget http://ftp.gnu.org/gnu/bc/bc-1.06.tar.gz
+  - tar xf bc-1.06.tar.gz
+  - (cd bc-1.06 && CC="ccache gcc" ./configure --quiet && make -j4)
```

Comment:
No, the cache does not start from scratch each time. There is one cache per branch, populated from master. See http://docs.travis-ci.com/user/caching/#Fetching-and-storing-caches.